### PR TITLE
feature(device): drop the device_tensor, torch-rbln checks 

### PR DESCRIPTION
**Pull request by rebel-eunji**
### 🚀 Summary of Changes

This PR removes the env var VLLM_RBLN_USE_DEVICE_TENSOR and checking installation of torch-rbln.
It removes compile_context clearly.

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [x] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -100,8 +100,8 @@ namespaces = false
...
     "model_compile: compiles a whole model on the NPU; minutes per test. Skipped unless --model-compile is passed. Tests that compile only an individual op are deliberately left unmarked so they stay in the default lane.",
-    "use_device: always touches the NPU (e.g. compiles an op via the rbln backend) regardless of --device-tensor, so it is always auto-spawned in a fresh process -- never run in the parent, which would pin the device for the rest of the session.",
-    "maybe_use_device: opens the RBLN device only when device tensors are on. Auto-spawned in a fresh process under --device-tensor 1 (never in the parent, which would pin the device for the rest of the session); runs in-process under --device-tensor 0 (cpu).",
+    "use_device: always touches the NPU (e.g. compiles an op via the rbln backend), so it is always auto-spawned in a fresh process -- never run in the parent, which would pin the device for the rest of the session.",
+    "maybe_use_device: opens the RBLN device when the platform's device_type is not cpu. Auto-spawned in a fresh process in that case (never in the parent, which would pin the device for the rest of the session); runs in-process otherwise.",
```
Comment (rebel-jonghewk) [inline pyproject.toml:104]: do we need to keep`maybe_use_device`?

**Review by rebel-jonghewk — COMMENTED** [review]
How about adding a deprecation warning? 
"VLLM_RBLN_USE_DEVICE_TENSOR" in os.environ: logger.warning(...)

**Review by rebel-jonghewk — APPROVED** [review]
Overall looks good. Can you check with the other teams (compiler, primitive) whether they need support for CPU tensors (fallbacks)? Then we can discuss when to merge.

File: pyproject.toml (hunk 1 above)
Comment (rebel-eunji) [reply pyproject.toml:104]: maybe not.. I'll remove it together.

Comment (rebel-cjlee) [reply pyproject.toml:104]: I think it could be my work. Refer to my mention above.
> There is still some leftover in the pytest setup that this PR does not touch. In tests/native, device_type is now always "rbln", so maybe_use_device does the same thing as use_device, and the device_type != "cpu" check in conftest.py is always true. I planned to clean this up long before this PR. I want to keep it out of this diff, so I will do it in a follow up PR once this one is merged.

Comment (rebel-eunji) [reply pyproject.toml:104]: I got it. Thanks.
